websites/C/Crunchyroll/presence.ts

```diff
@@ -109,14 +109,31 @@ presence.on("UpdateData", async () => {
   }
 
   if (iFrameVideo !== false && !isNaN(duration)) {
-    const videoTitle = document.querySelector(".ellipsis .text-link span"),
-      episod = document.querySelectorAll("#showmedia_about_media h4"),
-      epName = document.querySelector("h4#showmedia_about_name"),
-      episode = episod[1].innerHTML + " - " + epName.innerHTML,
-      timestamps = presence.getTimestamps(
-        Math.floor(currentTime),
-        Math.floor(duration)
-      );
+    let videoTitle,
+    type,
+    episode,
+    season;
+    if (document.location.hostname.startsWith("beta")) {
+      const episodeid = document.location.pathname.match(/watch\/(.*)\/.*/),
+      url = `https://beta-api.crunchyroll.com/cms/v2/US/M3/-/objects/${episodeid[1]}?Signature=A58Y6P76n7z72bcueov582lZiJcWCoJ-~AKn4y6khza30izYCquaIvRH9lhX6mY2goJpCyc9c74mI4QdWwYna8kfHUMiZmgD3ofvSGyZX88XTiR~nSVfNmBmRiGpPAR1MkRYe3pFWbAHOqPMNB89EKCg1Ho~LdecTbuQ3seOMip9QYYH6Z8T0GCFJlOxS3sMKT0hPak~nYHrj4La-IEmAe6PwbSxg59j3iIB7u8ft020CIkel68O~PjPL~MSot1eU7GZMGVQAkAOnKoZFLt~H2KeFQdHzsfp8Ns39jKNcvMwQ9BGU8XlEWQTBCNrWOycnTvafvBCPqFYXWMiTtiNXg__&Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly9iZXRhLWFwaS5jcnVuY2h5cm9sbC5jb20vY21zL3Y~L1VTL00zLy0vKiIsIkNvbmRpdGlvbiI6eyJEYXRlTGVzc1RoYW4iOnsiQVdTOkVwb2NoVGltZSI6MTYyMzA5MDU0M319fV19&Key-Pair-Id=APKAJMWSQ5S7ZB3MF5VA`,
+      obj = await (await fetch(url)).json(),
+      metadata = await obj.items[0],
+      episod = `E${metadata.episode_metadata.episode_number} - ${metadata.title}`;
+
+      type = metadata.episode_metadata.is_dubbed === true ? " (Dub)" : " (Sub)";
+      videoTitle = metadata.episode_metadata.series_title + type;
+      season = `S${metadata.episode_metadata.season_number}`;
+      episode = season + episod;
+    } else {
+      videoTitle = document.querySelector(".ellipsis .text-link span").innerHTML;
+      const episod = document.querySelectorAll("#showmedia_about_media h4"),
+        epName = document.querySelector("h4#showmedia_about_name");
+      episode = `${episod[1].innerHTML} - ${epName.innerHTML}`;
+    }
+    const timestamps = presence.getTimestamps(
```

Comment:
```suggestion
    const [, endTimestamp] = presence.getTimestamps(
```
and change `presenceData.endTimestamp = timestamps[1]` to `presenceData.endTimestamp = endTimestamp`